Source/Dafny/AST/DafnyAst.cs

```diff
@@ -1798,8 +1798,7 @@ public enum AutoInitInfo { MaybeEmpty, Nonempty, CompilableValue }
       Contract.Requires(type != null);
       type = type.NormalizeExpandKeepConstraints();
       List<Type> tower;
-      var udt = type as UserDefinedType;
-      if (udt != null && udt.ResolvedClass is SubsetTypeDecl) {
+      if (type is UserDefinedType udt && udt.ResolvedClass is SubsetTypeDecl) {
         var sst = (SubsetTypeDecl)udt.ResolvedClass;
```

Comment:
I think you can merge these two clauses together into a single pattern matching:
```suggestion
      if (type is UserDefinedType {ResolvedClass: SubsetTypeDecl sst} udt) {
```